ui/components/app/permission-connect-header/permission-connect-header.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  AlignItems,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+  Display,
+  BlockSize,
+  FontWeight,
+  FlexDirection,
+  BackgroundColor,
+} from '../../../helpers/constants/design-system';
+import {
+  IconSize,
+  Text,
+  Box,
+  AvatarFavicon,
+  AvatarBase,
+} from '../../component-library';
+
+const PermissionConnectHeader = ({ origin, iconUrl }) => {
+  const transformOriginToTitle = (rawOrigin) => {
+    try {
+      const url = new URL(rawOrigin);
+      const parts = url.hostname.split('.');
+      return parts.slice(-2).join('.');
+    } catch (e) {
+      return 'Unknown Origin';
+    }
+  };
+  const title = transformOriginToTitle(origin);
```

Comment:
To clarify this, we're using main domain name as the title and full domain (origin) for the subtitle. That was the original request. I think that we gave up on showing the website title, because it can be easy manipulated by malicious websites.

For example, here is how it looks when requesting from Snaps Directory:
![Screenshot 2024-04-11 at 12 37 01](https://github.com/MetaMask/metamask-extension/assets/13301024/59d8a836-6ef5-4f94-96d9-3acc223eb69b)

Reply:
Yep, the idea is that the title emphasises the domain (which is the most important part of the origin) and the subtitle shows the full URL